pymeasure/instruments/toptica/ibeamsmart.py

```diff
@@ -22,99 +22,192 @@
+        msg = []
+        try:
+            while True:
+                line = super().read()
+                if line == '[OK]':
+                    break
+                msg.append(line)
+        except VisaIOError:
+            reply = '\n'.join(msg)
+            try:
+                self.adapter.connection.flush_read_buffer()
+            except AttributeError:
+                log.warning("Adapter does not have 'flush_read_buffer' method.")
+            raise ValueError(
+                "TopticaAdapter.read(2): Error after command "
+                f"'{self.lastcommand}' with message '{reply}'")
+        reply = '\n'.join(msg)
+        r = self._reg_value.search(reply)
+        if r:
+            return r.groups()[0]
+        else:
+            return reply
+
+    def write(self, command):
+        self.lastcommand = command
+        super().write(command)
+
+    def check_errors(self):
+        """Check communication after setting a value.
+
+        Checks if the last reply is only '[OK]', otherwise a ValueError is
+        raised and the read buffer is flushed because one has to assume that
+        some communication is out of sync.
+        """
+        reply = self.read()
+        if reply:
+            # anything else than '[OK]'.
+            self.adapter.connection.flush()
+            raise ValueError(
+                f"IBeamSmart: Error after command '{self.lastcommand}' with "
+                f"message '{reply}'")
 
     version = Instrument.measurement(
-           "ver", """ Firmware version number """,
+        "ver", """Get Firmware version number.""",
     )
 
     serial = Instrument.measurement(
-           "serial", """ Serial number of the laser system """,
+        "serial", """Get Serial number of the laser system.""",
     )
 
     temp = Instrument.measurement(
-            "sh temp",
-            """ temperature of the laser diode in degree centigrade.""",
+        "sh temp",
+        """Measure the temperature of the laser diode in degree centigrade.""",
     )
 
     system_temp = Instrument.measurement(
-            "sh temp sys",
-            """ base plate (heatsink) temperature in degree centigrade.""",
+        "sh temp sys",
+        """Measure base plate (heatsink) temperature in degree centigrade.""",
     )
 
     laser_enabled = Instrument.control(
-            "sta la", "la %s",
-            """ Status of the laser diode driver.
-                This can be True if the laser is on or False otherwise""",
-            validator=strict_discrete_set,
-            values=[True, False],
-            get_process=lambda s: True if s == 'ON' else False,
-            set_process=lambda v: "on" if v else "off",
+        "sta la", "la %s",
+        """Control emission status of the laser diode driver (bool).""",
+        validator=strict_discrete_set,
+        values=[True, False],
+        get_process=lambda s: True if s == 'ON' else False,
+        set_process=lambda v: "on" if v else "off",
```

Comment:
```suggestion
        values={True: "on", False : "off"},
        map_values=True,
```
Is the suggested change is not possible because the instrument is using upper case in sending and lower case in reception ?

Reply:
I don't know the instrument itself. I just combined the previous "toptica adapter" and the instrument into one. Therefore I did not touch the properties (yet).

Reply:
Ok. If focus is replacing the adapter, probably my comment, if relevant, is not applicable.
I think it's a good habit to keep the focus to avoid PR too large. I would resolve this conversation then.

Reply:
I think the comment that this is not possible because of the specifics of the device is correct! There is a difference in sending and receiving (lower/upper case)